aviary/interface/methods_for_level2.py

```diff
@@ -346,127 +195,22 @@ def load_inputs(
+        self.problem_type = aviary_inputs.get_val(Settings.PROBLEM_TYPE)
 
-        if mission_method is TWO_DEGREES_OF_FREEDOM:
-            aviary_inputs.set_val(
-                Mission.Summary.CRUISE_MASS_FINAL,
-                val=self.initialization_guesses['cruise_mass_final'],
-                units='lbm')
-            aviary_inputs.set_val(
-                Mission.Summary.GROSS_MASS,
-                val=self.initialization_guesses['actual_takeoff_mass'],
-                units='lbm')
-
-            # Commonly referenced values
-            self.cruise_alt = aviary_inputs.get_val(
-                Mission.Design.CRUISE_ALTITUDE, units='ft')
-            self.problem_type = aviary_inputs.get_val(Settings.PROBLEM_TYPE)
-            self.mass_defect = aviary_inputs.get_val('mass_defect', units='lbm')
-
-            self.cruise_mass_final = aviary_inputs.get_val(
-                Mission.Summary.CRUISE_MASS_FINAL, units='lbm')
-
-            if self.post_mission_info is True and 'target_range' in self.post_mission_info:
-                self.target_range = wrapped_convert_units(
-                    phase_info['post_mission']['target_range'], 'NM')
-                aviary_inputs.set_val(Mission.Summary.RANGE,
-                                      self.target_range, units='NM')
-            else:
-                self.target_range = aviary_inputs.get_val(
-                    Mission.Design.RANGE, units='NM')
-                aviary_inputs.set_val(Mission.Summary.RANGE, aviary_inputs.get_val(
-                    Mission.Design.RANGE, units='NM'), units='NM')
-            self.cruise_mach = aviary_inputs.get_val(Mission.Design.MACH)
-            self.require_range_residual = True
-
-        elif mission_method is HEIGHT_ENERGY:
-            self.problem_type = aviary_inputs.get_val(Settings.PROBLEM_TYPE)
-            aviary_inputs.set_val(
-                Mission.Summary.GROSS_MASS,
-                val=self.initialization_guesses['actual_takeoff_mass'],
-                units='lbm')
-            if 'target_range' in self.post_mission_info:
-                aviary_inputs.set_val(Mission.Summary.RANGE, wrapped_convert_units(
-                    phase_info['post_mission']['target_range'], 'NM'), units='NM')
-                self.require_range_residual = True
-                self.target_range = wrapped_convert_units(
-                    phase_info['post_mission']['target_range'], 'NM')
-            else:
-                self.require_range_residual = False
-                # still instantiate target_range because it is used for default guesses
-                # for phase comps
-                self.target_range = aviary_inputs.get_val(
-                    Mission.Design.RANGE, units='NM')
+        self.builder.initial_guesses(self)
+        # This function sets all the following defaults if they were not already set
+        # self.engine_builders, self.mass_method, self.pre_mission_info, self_post_mission_info
```

Comment:
Looks like mass_method is not dealt with in here after all, we deal with it further up

Reply:
Thanks, I have removed it.